lib/rules/no-implied-eval.js

```diff
@@ -28,141 +34,130 @@ module.exports = {
-         *
-         * This checks the callee name and that there's an argument, but not the type of the argument.
-         * @param {ASTNode} node The CallExpression to check.
-         * @returns {boolean} True if the node matches, false if not.
-         * @private
+         * Checks whether a node is a Literal node which has one of the specified values.
+         * @param {ASTNode} node A node to check.
+         * @param {string[]} specifiers Array of specified value.
+         * @returns {boolean} True if the node is a Literal node which has one of the specified values.
          */
-        function isImpliedEvalCallExpression(node) {
-            const isMemberExpression = (node.callee.type === "MemberExpression"),
-                isIdentifier = (node.callee.type === "Identifier"),
-                isImpliedEvalCallee =
-                    (isIdentifier && CALLEE_RE.test(node.callee.name)) ||
-                    (isMemberExpression && isImpliedEvalMemberExpression(node.callee));
-
-            return isImpliedEvalCallee && node.arguments.length;
+        function isSpecifiedConstant(node, specifiers) {
+            switch (node.type) {
+                case "Literal":
+                    return specifiers.includes(node.value);
+
+                case "TemplateLiteral":
+                    return (
+                        node.expressions.length === 0 &&
+                        specifiers.includes(node.quasis[0].value.cooked)
+                    );
+
+                default:
+                    return false;
+            }
         }
 
         /**
-         * Checks that the parent is a direct descendent of an potential implied eval CallExpression, and if the parent is a CallExpression, that we're the first argument.
-         * @param {ASTNode} node The node to inspect the parent of.
-         * @returns {boolean} Was the parent a direct descendent, and is the child therefore potentially part of a dangerous argument?
-         * @private
+         * Checks a given node is a MemberExpression node which has the specified name's
+         * property.
+         * @param {ASTNode} node A node to check.
+         * @param {string[]} specifiers Array of specified name.
+         * @returns {boolean} `true` if the node is a MemberExpression node which has
+         *      the specified name's property
          */
-        function hasImpliedEvalParent(node) {
-
-            // make sure our parent is marked
-            return node.parent === last(last(impliedEvalAncestorsStack)) &&
-
-                // if our parent is a CallExpression, make sure we're the first argument
-                (node.parent.type !== "CallExpression" || node === node.parent.arguments[0]);
+        function isSpecifiedMember(node, specifiers) {
+            return (
+                node.type === "MemberExpression" &&
+                (node.computed ? isSpecifiedConstant : isSpecifiedIdentifier)(node.property, specifiers)
+            );
```

Comment:
Non-blocker for this PR, could this be simplified with:

```js
return node.type === "MemberExpression" && specifiers.includes(astUtils.getStaticPropertyName(node));
```

so `function isSpecifiedConstant` wouldn't be necessary.